extractor/filesystem/language/python/internal/pypipurl/pythonpurl.go

```diff
@@ -0,0 +1,22 @@
+// Package pypipurl converts an inventory to a PyPI type PackageURL.
+package pypipurl
+
+import (
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/purl"
+)
+
+// MakePackageURL returns a package URL following the purl PyPI spec:
+// - Name is lowercased
+// - Replaces _ with -
+//
+// See: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#pypi
+func MakePackageURL(i *extractor.Inventory) *purl.PackageURL {
+	return &purl.PackageURL{
+		Type:    purl.TypePyPi,
+		Name:    strings.ReplaceAll(strings.ToLower(i.Name), "_", "-"),
```

Comment:
The PURL spec is lacking here, but I think we need to go a bit further (https://peps.python.org/pep-0503/#normalized-names) and replace "all runs of the characters ., -, or _ replaced with a single - character". 

This makes it consistent with the OSV schema as well.

Reply:
Updated this to follow the PEP standard, and added some tests.